stripe-core/src/main/java/com/stripe/android/core/strings/ResolvableStringUtils.kt

```diff
@@ -0,0 +1,64 @@
+package com.stripe.android.core.strings
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+import androidx.annotation.StringRes
+
+internal fun resolveArgs(context: Context, args: Array<out Any?>): Array<Any?> {
+    return args.map { arg ->
+        when (arg) {
+            is ResolvableString -> arg.resolve(context)
+            else -> arg
+        }
+    }.toTypedArray()
+}
+
+/**
+ * Creates a `ResolvableString` from an identifier.
+ *
+ * @param id a string resource identifier
+ *
+ * @return a `ResolvableString` instance when resolved returns the string that matches the identifier.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun resolvableString(@StringRes id: Int): ResolvableString {
+    return IdentifierResolvableString(id, arrayOf())
+}
+
+/**
+ * Creates a `ResolvableString` from a given identifier.
+ *
+ * @param id a string resource identifier
+ * @param formatArgs a variable amount of arguments to format the string with.
+ *
+ * @return a `ResolvableString` instance when resolved returns the string that matches the identifier.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun resolvableString(@StringRes id: Int, vararg formatArgs: Any?): ResolvableString {
+    return IdentifierResolvableString(id, formatArgs)
+}
+
+/**
+ * Creates a `ResolvableString` from a given string value.
+ *
+ * @param value a string value
+ *
+ * @return a `ResolvableString` instance when resolved returns the string value.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun resolvableString(value: String): ResolvableString {
+    return StaticResolvableString(value, arrayOf())
+}
+
+/**
+ * Creates a `ResolvableString` from a given string value.
+ *
+ * @param value a string value
+ * @param formatArgs a variable amount of arguments to format the string with.
+ *
+ * @return a `ResolvableString` instance when resolved returns the string value.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun resolvableString(value: String, vararg formatArgs: Any?): ResolvableString {
```

Comment:
I believe so yes! This is useful in potential scenarios where we use a mix of string values & string resources. For example, a nullable string value from our data source in which we also have a default string resource when it is null. The consumer wouldn't know since it only recognizes it as a `ResolvableString`.